test/integration/puppetlabs/services/master/environment_classes_int_test.clj

```diff
@@ -458,3 +480,43 @@
                   (response->class-info-map
                    test-response-after-all-flush))
                "unexpected body for test response")))))))
+
+(deftest ^:integration
+         not-modified-returned-for-environment-class-info-request-with-gzip-tag
+  (testing (str "SERVER-1153 - when the webserver gzips the response "
+                "containing environment_classes etag, the next request "
+                "roundtripping that etag returns an HTTP 304 (Not Modified)")
+    (let [expected-etag "abcd1234"
+          body-length 200000
+          jruby-service (reify jruby/JRubyPuppetService
+                          (get-environment-class-info-tag [_ _]
+                            expected-etag))
+          app (->
+               (fn [_]
+                 (master-core/response-with-etag
+                  (apply str (repeat body-length "a"))
+                  expected-etag))
+               (master-core/wrap-with-etag-check jruby-service))]
+      (jetty9/with-test-webserver
+       app
+       port
+       (let [request-url (str "http://localhost:" port)
+             initial-response (http-client/get request-url {:as :text})
+             response-tag (response-etag initial-response)
```

Comment:
yes, the test is fine as it is, just thought maybe the extra assertion would have some value in the future.